src/connection/mc_auth_logic.erl

```diff
@@ -125,8 +125,7 @@ generate_sig(SaltedPassword, AuthMessage) ->
 
 %% @private
 hi(Password, Salt, Iterations) ->
-  {ok, Key} = pbkdf2:pbkdf2(sha, Password, Salt, Iterations, 20),
-  Key.
+  crypto:pbkdf2_hmac(sha, Password, Salt, Iterations, 20).
```

Comment:
We probably want to follow the pattern in #232 to avoid breaking the library on older releases of OTP. That is, `crypto:pbkdf2_hmac` is new in OTP 24 and missing from older versions, so if we run this path on OTP 23 we'll crash for the same reason this fix exists.

Checking for OTP version
https://github.com/dmsnell/mongodb-erlang/blob/aa28246729086681239ccccfd91db5baafdc5b3f/src/support/mc_utils.erl#L12-L18

Conditionally using newer functions
https://github.com/dmsnell/mongodb-erlang/blob/aa28246729086681239ccccfd91db5baafdc5b3f/src/support/mc_utils.erl#L72-L76

Reply:
updated, didn't know crypto in OTP < 24 doesn't have pbkdf2_hmac, thanks for the info

Reply:
My read on [the docs](https://www.erlang.org/doc/man/crypto.html#pbkdf2_hmac-5) is that it came in with OTP 24.2, with [crypto 5.0.5](https://www.erlang.org/patches/otp-24.2#crypto-5.0.5)

Come to think of it, do we need to check for 24.2 even? I guess OTP 24.0 and OTP 24.1 won't have it either. I'm not sure off-hand how to check for the minor version and I hope we can do it in the macro so this doesn't have to be a runtime check.

Reply:
I couldn't find a predefined macro returns the minor/patch version